wrappers/python/tests/layer/test_llama_attention.py

```diff
@@ -39,37 +37,77 @@ class LlamaAttentionTestParams:
     n_head_kv: int
     dtype: np.dtype
     bias: bool
-    layer_idx: int=0
+    layer_idx: int = 0
+
 
-TEST_PARAMS = \
-[ \
-    LlamaAttentionTestParams(n_emb=128, n_emb_tile=32, n_seq=64, \
-        n_seq_tile=16, n_batch=4, n_batch_tile=1, n_head=16, n_head_tile=8, \
-        n_head_kv=4, dtype=np.dtypes.Float32DType(), bias=True), \
-    LlamaAttentionTestParams(n_emb=128, n_emb_tile=32, n_seq=64, \
-        n_seq_tile=16, n_batch=4, n_batch_tile=1, n_head=16, n_head_tile=8, \
-        n_head_kv=4, dtype=np.dtypes.Float32DType(), bias=False), \
-    LlamaAttentionTestParams(n_emb=128, n_emb_tile=128, n_seq=64, \
-        n_seq_tile=64, n_batch=3, n_batch_tile=3, n_head=8, n_head_tile=4, \
-        n_head_kv=4, dtype=np.dtypes.Float32DType(), bias=True), \
-    LlamaAttentionTestParams(n_emb=128, n_emb_tile=128, n_seq=64, \
-        n_seq_tile=64, n_batch=3, n_batch_tile=3, n_head=8, n_head_tile=4, \
-        n_head_kv=4, dtype=np.dtypes.Float32DType(), bias=False), \
+TEST_PARAMS = [
+    LlamaAttentionTestParams(
+        n_emb=128,
+        n_emb_tile=32,
+        n_seq=64,
+        n_seq_tile=16,
+        n_batch=4,
+        n_batch_tile=1,
+        n_head=16,
+        n_head_tile=8,
+        n_head_kv=4,
+        dtype=np.dtypes.Float32DType(),
+        bias=True,
+    ),
+    LlamaAttentionTestParams(
+        n_emb=128,
+        n_emb_tile=32,
+        n_seq=64,
+        n_seq_tile=16,
+        n_batch=4,
+        n_batch_tile=1,
+        n_head=16,
+        n_head_tile=8,
+        n_head_kv=4,
+        dtype=np.dtypes.Float32DType(),
```

Comment:
Just `np.float32`.

Reply:
I just repeated how it is done in a batch norm test. We have to either use Torch dtypes, or rely on our own dtypes. Supporting bf16, fp16, fp8 and quantised formats is out of Numpy scope.

Reply:
#102 is to resolve this issue in the future

Reply:
> Supporting bf16, fp16, fp8 and quantised formats is out of Numpy scope.

Absolutely not. There are widely-adopted custom types (e.g. `pandas`). Also, NumPy facilitates an extensions with [NEP-42][1]. Thanks to JAX team, we have [ml-dtypes][1] which provide in framework-agnostic way common low-bits floating types.

\[1\] [NEP-42][1]
\[2\] [numpy/numpy-user-dtypes][2]
\[3\] [jax-ml/ml-dtypes][3]

[1]: https://numpy.org/neps/nep-0042-new-dtypes.html
[2]: https://github.com/numpy/numpy-user-dtypes
[3]: https://github.com/jax-ml/ml_dtypes

Reply:
NEP-42 is yet under construction, as the link says, but in the future it will allow us to add our own quantised formats. JAX ml-dtypes can be used for now, as it covers bfloat16 and float8 types. We need these only for testing purposes, as an interoperation between NNTile and Numpy is through upcasting NNTile data into a `float` for all 32 and less bit floats and into a `double` for 64-bit floats.

Reply:
> We need these only for testing purposes, as an interoperation between NNTile and Numpy is through upcasting NNTile data into a float for all 32 and less bit floats and into a double for 64-bit floats.

Then we do not actually need a dtypes. Just define `upcast()` routine and expose it to Python with limited visiblity. That's all.

Reply:
This is a distinct topic. I don't understand why you brought it now.

Original comment was about replacing internal `numpy.dtypes.*` in favor of `numpy.float32`.

Reply:
I just made it a string. Looks much simpler now